lib/svg/svg.py

```diff
@@ -14,20 +14,13 @@ def get_document(node):
     return node.getroottree().getroot()
 
 
-def generate_unique_id(document_or_element, prefix="path"):
+def generate_unique_id(document_or_element, prefix="path", blacklist=None):
     if isinstance(document_or_element, etree._ElementTree):
         document = document_or_element.getroot()
     else:
         document = get_document(document_or_element)
 
-    doc_ids = {node.get('id') for node in document.iterdescendants() if 'id' in node.attrib}
-
-    i = 1
-    while True:
-        new_id = "%s%d" % (prefix, i)
-        if new_id not in doc_ids:
-            break
-        i += 1
+    new_id = document.get_unique_id(prefix, blacklist=blacklist)
```

Comment:
I'm guessing this PR isn't done yet?  What are you going to pass for the blocklist argument?

Reply:
Not sure what a blocklist would be?
It's blacklisting ids that are used previously before adding to document and might need to be blacklisted...
It's just there for possible later use cases so we know it is possible. Should I take it out?
https://inkscape.gitlab.io/extensions/documentation/source/inkex.elements._svg.html?highlight=unique_id#inkex.elements._svg.SvgDocumentElement.get_unique_id